### ISSUE TITLE
Where in the code does it reflect the role of the relational-aware map M-Zot?

### ISSUE THREAD

**Issue by CyWang-Jade** [Kebii/Zoom-Transformer#1]
Dear author, thank you very much for your contributions in the field of action recognition. While going through the code of your article "Zoom," I am confused about the Attention module in the Transformer. It seems that you did not utilize the code when the mask is not None, which implies that M-Zot did not play a role. The relevant code is shown below, where the addition of dots and mask does not seem to be used in the subsequent code.

```
class Attention(nn.Module):
    def __init__(self, dim, out_dim, heads = 3, dropout = 0.1):
        super().__init__()
        self.heads = heads
        self.scale = dim ** -0.5  # 1/sqrt(dim)

        self.to_qkv = nn.Linear(dim, dim*3, bias = True)
        torch.nn.init.xavier_uniform_(self.to_qkv.weight)
        torch.nn.init.zeros_(self.to_qkv.bias)
        
        self.nn1 = nn.Linear(dim, out_dim)
        torch.nn.init.xavier_uniform_(self.nn1.weight)
        torch.nn.init.zeros_(self.nn1.bias)        
        self.do1 = nn.Dropout(dropout)
        

    def forward(self, x, mask = None):
        b, n, _, h = *x.shape, self.heads
        qkv = self.to_qkv(x) #gets q = Q = Wq matmul x1, k = Wk mm x2, v = Wv mm x3
        q, k, v = rearrange(qkv, 'b n (qkv h d) -> qkv b h n d', qkv = 3, h = h) # split into multi head attentions

        dots = torch.einsum('bhid,bhjd->bhij', q, k) * self.scale

        attn = dots.softmax(dim=-1) #follow the softmax,q,d,v equation in the paper

        if mask is not None:
            assert mask.shape[-1] == dots.shape[-1], 'mask has incorrect dimensions'
            dots = (dots + mask)*0.5  #"Dots" contains the M-Zot matrix, but it appears that it is not utilized in the subsequent code.

        out = torch.einsum('bhij,bhjd->bhid', attn, v) #product of v times whatever inside softmax
        out = rearrange(out, 'b h n d -> b n (h d)') #concat heads into one matrix, ready for next encoder block
        out =  self.nn1(out)
        out = self.do1(out)

        return out
```

**Comment by Kebii** [comment]
Thanks for your attention to our work. 
I have checked the code and found that this problem does exist in this version. I'm sorry for mistakenly uploading the code of the ablation experiment for the R-A Map.
You can get the correct code by adding "attn = dots.softmax(dim=-1)" after the "dots".
